FIX: Multiple dates for classes

zachomedia opened the pull request:
Fixed an issue where classes with multiple dates would not display
correct. Each class now has a child array 'classes'.

zachomedia committed:
FIX: Multiple dates for classes

Fixed an issue where classes with multiple dates would not display
correct. Each class now has a child array 'classes'.

KartikTalwar commented:
Hey @zachomedia. This is awesome!

This mostly represents everything from the schedule page, except a slight change in how classes with multiple dates show up if that class has a reserve. Testing [this course](http://www.adm.uwaterloo.ca/cgi-bin/cgiwrap/infocour/salook.pl?sess=1139&subject=PHYS&cournum=131L):

``` php
$data = parse_schedule('1139', 'PHYS', '131L');
print_r($data[1]); // course_id => 6347
```

Whats happening for this case is, when the reserve row is read, the data after `waiting_total` is discarded, instead of being pushed to `$new_class`. Therefore, instead of showing 5 timeslots, it ends up showing only 4.

zachomedia committed:
Fixes: (1) Classes weren't listed if reserves were on the same line, (2) Classes were listed in revers order. They now appear in the correct order.

Fixes:
(1) Classes weren't listed if reserves were on the same line.

(2) Classes were listed in revers order. They now appear in the correct
order.

zachomedia commented:
@KartikTalwar 

Thanks for catching that! I have committed an update which should resolve it. In addition, I also fixed the order of the multiple date/times for a class, as they were appearing in reverse order.
